### DISCUSSION TITLE
Handling of large Codelists (ItemSchemes in general?) - Pagination?

### DISCUSSION THREAD

**Discussion by sosna** [sdmx-twg/sdmx-rest#175] [General]
(Was initially posted by @buhaiovos in the Issues list)

Hello!

Preemptively sorry in case this is a wrong communication channel. Are there any plans or maybe some approaches that already exist to handle queries for large codelists? Let's say we have a client-app that due to some limitations has issues dealing with a codelist that has a couple of millions of codes. Are there any "right" ways to implement partial request for a portion of a codelist to do some kind of pagination/lazy load of the codelist's contents?

Thanks!

**Comment by sosna** [comment]
This is not supported at the moment.

However, one of the open issues is related to pagination (cf. issue #50). While this was initially about data queries, we will consider your use case within this context too.

Out of curiosity, how large (i.e. how many items) are your codelists?

**Reply by buhaiovos** [reply]
Sorry for a late reply. Thanks for the link!
So, currently we know for sure that we would have to deal with codelists which have ~300k codes. But in theory, we would need to support 1M case as well.

**Reply by sosna** [reply]
Perfect, thanks! I'll add this info to the respective issue.
